share/functions/__fish_describe_command.fish

```diff
@@ -6,15 +6,34 @@
 # The whatis database is non-existent, so apropos tries (and fails) to create it every time,
 # which takes about half a second.
 #
-# So we disable this entirely in that case.
-if test (uname) = Darwin
-    set -l darwin_version (uname -r | string split .)
-    # macOS 15 is Darwin 19, this is an issue up to and including 10.15.3.
-    if test "$darwin_version[1]" = 19 -a "$darwin_version[2]" -le 3
-        function __fish_describe_command
+# Instead, we build a whatis database in the user cache directory
+# and override the MANPATH using that directory before we run `apropos`
+#
+# the cache is rebuilt one a day
+#
+function __fish_apropos
+    if test (uname) = Darwin
```

Comment:
You can probably also remove the Darwin workaround in `share/functions/__fish_complete_man.fish`.

---

I wonder if this should just be a full wrapper around apropos
`share/functions/apropos.fish` could start like

```fish
if not test (uname) = Darwin
    exit
end

function apropos
    ...
end
```

so we only override it on MacOS.

I don't really know, either way is fine by me. If we expect that a user would always prefer our version, than the wrapper is a good idea.

Reply:
Thought about that as well.

It definitely makes sense, since the wrapper will be able to find more as well.

I can make a similar override for `whatis` using `MANPAGE=$cache`

What do you think @faho?

Reply:
I've just pushed the following changes:
* created `functions/__fish_apropos.fish`
* changed `__fish_complete_man` to use `__fish_apropos` instead